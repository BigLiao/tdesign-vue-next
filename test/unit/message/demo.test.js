import { mount } from '@vue/test-utils';

// unit test for component in examples.
describe('Message', () => {
  it('type demo works fine', () => {
    const demo = require('@/examples/message/demos/type.vue');
    const wrapper = mount(demo);
    expect(wrapper.element).toMatchSnapshot();
  });
  // it('close demo works fine', () => {
  //   const demo = require('@/examples/message/demos/close.vue');
  //   const wrapper = mount(demo);
  //   expect(wrapper).toMatchSnapshot();
  // });
  // it('closeAll demo works fine', () => {
  //   const demo = require('@/examples/message/demos/closeAll.vue');
  //   const wrapper = mount(demo);
  //   expect(wrapper.element).toMatchSnapshot();
  // });
  // it('offset demo works fine', () => {
  //   const demo = require('@/examples/message/demos/offset.vue');
  //   const wrapper = mount(demo);
  //   expect(wrapper.element).toMatchSnapshot();
  // });
  // it('placement demo works fine', () => {
  //   const demo = require('@/examples/message/demos/placement.vue');
  //   const wrapper = mount(demo);
  //   expect(wrapper.element).toMatchSnapshot();
  // });
  // it('plugin demo works fine', () => {
  //   const demo = require('@/examples/message/demos/plugin.vue');
  //   const wrapper = mount(demo);
  //   expect(wrapper.element).toMatchSnapshot();
  // });
  // it('toggle demo works fine', () => {
  //   const demo = require('@/examples/message/demos/toggle.vue');
  //   const wrapper = mount(demo);
  //   expect(wrapper.element).toMatchSnapshot();
  // });
});
