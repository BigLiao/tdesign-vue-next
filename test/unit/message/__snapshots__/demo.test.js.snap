// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message close demo works fine 1`] = `
<div
  class="t-demo-message"
>
  <div
    class="t-message t-is-info t-is-closable"
  >
    <svg
      class="t-icon t-icon-info-circle-filled"
      version="1.1"
      viewBox="0 0 1024 1024"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M512 96C741.7504576 96 928 282.24954432 928 512 928 741.7504576 741.7504576 928 512 928 282.24954432 928 96 741.7504576 96 512 96 282.24954432 282.24954432 96 512 96ZM544 416L416 416 416 480 480 480 480 736 608 736 608 672 544 672 544 416ZM544 288L480 288 480 352 544 352 544 288Z"
        fill-opacity=".9"
        fill-rule="evenodd"
      />
    </svg>
    
    点击关闭按钮触发事件 click-close-btn 
    
    <span
      class="t-message-close"
    >
      <svg
        class="t-icon t-icon-close"
        version="1.1"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M512.0032 570.83072L708.864 767.6864 767.6928 708.8576 570.83456 511.99936 767.6928 315.14176 708.864 256.3104 512.0032 453.168 315.14368 256.30848 256.31232 315.13984 453.17184 511.99936 256.31232 708.8576 315.14368 767.6928 512.0032 570.83072Z"
        />
      </svg>
    </span>
  </div>
  <div
    class="t-message t-is-info t-is-closable"
  >
    <svg
      class="t-icon t-icon-info-circle-filled"
      version="1.1"
      viewBox="0 0 1024 1024"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M512 96C741.7504576 96 928 282.24954432 928 512 928 741.7504576 741.7504576 928 512 928 282.24954432 928 96 741.7504576 96 512 96 282.24954432 282.24954432 96 512 96ZM544 416L416 416 416 480 480 480 480 736 608 736 608 672 544 672 544 416ZM544 288L480 288 480 352 544 352 544 288Z"
        fill-opacity=".9"
        fill-rule="evenodd"
      />
    </svg>
    
    自定义关闭按钮（文字）
    
    <span
      class="t-message-close"
    >
      关闭
    </span>
  </div>
  <div
    class="t-message t-is-info t-is-closable"
  >
    <svg
      class="t-icon t-icon-info-circle-filled"
      version="1.1"
      viewBox="0 0 1024 1024"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M512 96C741.7504576 96 928 282.24954432 928 512 928 741.7504576 741.7504576 928 512 928 282.24954432 928 96 741.7504576 96 512 96 282.24954432 282.24954432 96 512 96ZM544 416L416 416 416 480 480 480 480 736 608 736 608 672 544 672 544 416ZM544 288L480 288 480 352 544 352 544 288Z"
        fill-opacity=".9"
        fill-rule="evenodd"
      />
    </svg>
    
    自定义关闭按钮（函数）
    
    <span
      class="t-message-close"
    >
      <div
        class="t-message-close"
      >
        <b>
          x
        </b>
      </div>
    </span>
  </div>
  <div
    class="t-message t-is-info t-is-closable"
  >
    <svg
      class="t-icon t-icon-info-circle-filled"
      version="1.1"
      viewBox="0 0 1024 1024"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M512 96C741.7504576 96 928 282.24954432 928 512 928 741.7504576 741.7504576 928 512 928 282.24954432 928 96 741.7504576 96 512 96 282.24954432 282.24954432 96 512 96ZM544 416L416 416 416 480 480 480 480 736 608 736 608 672 544 672 544 416ZM544 288L480 288 480 352 544 352 544 288Z"
        fill-opacity=".9"
        fill-rule="evenodd"
      />
    </svg>
    
     自定义关闭按钮（插槽） 
    
    <span
      class="t-message-close"
    >
      
      <div
        class="t-message-close"
      >
        <b>
          x
        </b>
      </div>
      
    </span>
  </div>
</div>
`;

exports[`Message closeAll demo works fine 1`] = `
<div>
  <button
    class="t-button t-size-m t-button--variant-outline t-button--theme-default"
    type="button"
  >
    <span
      class="t-button__text"
    >
      
      点击打开多个消息
      
    </span>
  </button>
  <button
    class="t-button t-size-m t-button--variant-outline t-button--theme-default"
    id="t-demo-msg-close-all"
    type="button"
  >
    <span
      class="t-button__text"
    >
      
      点击关闭所有消息
      
    </span>
  </button>
</div>
`;

exports[`Message offset demo works fine 1`] = `
<div>
  <p>
     偏移量 offset 是相对于 placement 的偏移量，默认值：空，参数示例：{left, top, bottom, right}。 
  </p>
  <br />
  <button
    class="t-button t-size-m t-button--variant-base t-button--theme-primary"
    type="button"
  >
    <span
      class="t-button__text"
    >
      
      带 offset 信息位置
      
    </span>
  </button>
</div>
`;

exports[`Message placement demo works fine 1`] = `
<div>
  <div
    class="t-message-offset"
  >
    <div
      class="t-input t-size-m"
      modelvalue=""
    >
      <!---->
      <input
        autocomplete="false"
        class="t-input__inner"
        placeholder="请输入横向偏移量"
        type="text"
      />
      <!---->
    </div>
    <div
      class="t-input t-size-m"
      modelvalue=""
    >
      <!---->
      <input
        autocomplete="false"
        class="t-input__inner"
        placeholder="请输入纵向偏移量"
        type="text"
      />
      <!---->
    </div>
  </div>
  <div
    class="tdesign-tooltip-placement"
  >
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-default placement-center"
      type="button"
    >
      <span
        class="t-button__text"
      >
        
        center
        
      </span>
    </button>
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-default placement-top"
      type="button"
    >
      <span
        class="t-button__text"
      >
        
        top
        
      </span>
    </button>
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-default placement-left"
      type="button"
    >
      <span
        class="t-button__text"
      >
        
        left
        
      </span>
    </button>
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-default placement-right"
      type="button"
    >
      <span
        class="t-button__text"
      >
        
        right
        
      </span>
    </button>
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-default placement-bottom"
      type="button"
    >
      <span
        class="t-button__text"
      >
        
        bottom
        
      </span>
    </button>
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-default placement-top-left"
      type="button"
    >
      <span
        class="t-button__text"
      >
        
        top-left
        
      </span>
    </button>
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-default placement-top-right"
      type="button"
    >
      <span
        class="t-button__text"
      >
        
        top-right
        
      </span>
    </button>
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-default placement-bottom-left"
      type="button"
    >
      <span
        class="t-button__text"
      >
        
        bottom-left
        
      </span>
    </button>
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-default placement-bottom-right"
      type="button"
    >
      <span
        class="t-button__text"
      >
        
        bottom-right
        
      </span>
    </button>
  </div>
</div>
`;

exports[`Message plugin demo works fine 1`] = `
<div>
  <p>
    插件调用
  </p>
  <br />
  <div
    class="t-demo-message-duration"
  >
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-primary"
      type="button"
    >
      <span
        class="t-button__text"
      >
        
        消息
        
      </span>
    </button>
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-success"
      type="button"
    >
      <span
        class="t-button__text"
      >
        
         成功 
        
      </span>
    </button>
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-warning"
      type="button"
    >
      <span
        class="t-button__text"
      >
        
        警示
        
      </span>
    </button>
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-danger"
      type="button"
    >
      <span
        class="t-button__text"
      >
        
         失败 
        
      </span>
    </button>
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-primary"
      type="button"
    >
      <span
        class="t-button__text"
      >
        
        询问
        
      </span>
    </button>
    <!-- 0 表示永远不自动消失 -->
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-primary"
      type="button"
    >
      <span
        class="t-button__text"
      >
        
        加载中
        
      </span>
    </button>
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-default"
      type="button"
    >
      <span
        class="t-button__text"
      >
        
        关闭所有
        
      </span>
    </button>
  </div>
  <br />
  <br />
  <p>
    函数式调用
  </p>
  <br />
  <div
    class="t-demo-message-theme"
  >
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-primary"
      type="button"
    >
      <span
        class="t-button__text"
      >
        
        消息
        
      </span>
    </button>
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-success"
      type="button"
    >
      <span
        class="t-button__text"
      >
        
        成功
        
      </span>
    </button>
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-warning"
      type="button"
    >
      <span
        class="t-button__text"
      >
        
         警示 
        
      </span>
    </button>
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-danger"
      type="button"
    >
      <span
        class="t-button__text"
      >
        
        失败
        
      </span>
    </button>
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-primary"
      type="button"
    >
      <span
        class="t-button__text"
      >
        
        询问
        
      </span>
    </button>
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-primary"
      type="button"
    >
      <span
        class="t-button__text"
      >
        
        加载中
        
      </span>
    </button>
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-default"
      type="button"
    >
      <span
        class="t-button__text"
      >
        
        关闭所有
        
      </span>
    </button>
  </div>
  <br />
  <br />
</div>
`;

exports[`Message toggle demo works fine 1`] = `
<div
  id="t-demo-message-toggle"
>
  <button
    class="t-button t-size-m t-button--variant-outline t-button--theme-default"
    type="button"
  >
    <span
      class="t-button__text"
    >
      
       自由控制关闭时机（打开） 
      
    </span>
  </button>
</div>
`;

exports[`Message type demo works fine 1`] = `
<div
  class="t-demo-message"
>
  <div
    class="t-message t-is-info"
  >
    <svg
      class="t-icon t-icon-info-circle-filled"
      version="1.1"
      viewBox="0 0 1024 1024"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M512 96C741.7504576 96 928 282.24954432 928 512 928 741.7504576 741.7504576 928 512 928 282.24954432 928 96 741.7504576 96 512 96 282.24954432 282.24954432 96 512 96ZM544 416L416 416 416 480 480 480 480 736 608 736 608 672 544 672 544 416ZM544 288L480 288 480 352 544 352 544 288Z"
        fill-opacity=".9"
        fill-rule="evenodd"
      />
    </svg>
    
    用户表示普通操作信息提示
    
    <span
      class="t-message-close"
    >
      <!---->
    </span>
  </div>
  <div
    class="t-message t-is-success"
  >
    <svg
      class="t-icon t-icon-check-circle-filled"
      version="1.1"
      viewBox="0 0 1024 1024"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M512 96C741.7504576 96 928 282.24954432 928 512 928 741.7504576 741.7504576 928 512 928 282.24954432 928 96 741.7504576 96 512 96 282.24954432 282.24954432 96 512 96ZM690.8655168 353.58818368L464.56401024 579.82818368 328.82684416 444.09785152 283.57201024 489.35268544 464.59134592 670.3720192 509.84617984 625.1171872 736.120352 398.8430176 690.8655168 353.58818368Z"
        fill-opacity=".9"
        fill-rule="evenodd"
      />
    </svg>
    
    用户表示操作顺利达成
    
    <span
      class="t-message-close"
    >
      <!---->
    </span>
  </div>
  <div
    class="t-message t-is-warning"
  >
    <svg
      class="t-icon t-icon-error-circle-filled"
      version="1.1"
      viewBox="0 0 1024 1024"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M512 96C741.7504576 96 928 282.24954432 928 512 928 741.7504576 741.7504576 928 512 928 282.24954432 928 96 741.7504576 96 512 96 282.24954432 282.24954432 96 512 96ZM544 672L480 672 480 736 544 736 544 672ZM544 288L480 288 480 608 544 608 544 288Z"
        fill-opacity=".9"
        fill-rule="evenodd"
      />
    </svg>
    
    用户表示操作引起一定后果
    
    <span
      class="t-message-close"
    >
      <!---->
    </span>
  </div>
  <div
    class="t-message t-is-error"
  >
    <svg
      class="t-icon t-icon-error-circle-filled"
      version="1.1"
      viewBox="0 0 1024 1024"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M512 96C741.7504576 96 928 282.24954432 928 512 928 741.7504576 741.7504576 928 512 928 282.24954432 928 96 741.7504576 96 512 96 282.24954432 282.24954432 96 512 96ZM544 672L480 672 480 736 544 736 544 672ZM544 288L480 288 480 608 544 608 544 288Z"
        fill-opacity=".9"
        fill-rule="evenodd"
      />
    </svg>
    
    用户表示操作引起严重的后果
    
    <span
      class="t-message-close"
    >
      <!---->
    </span>
  </div>
  <div
    class="t-message t-is-question"
  >
    <svg
      class="t-icon t-icon-help-circle-filled"
      version="1.1"
      viewBox="0 0 1024 1024"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M512 96C741.7504576 96 928 282.24954432 928 512 928 741.7504576 741.7504576 928 512 928 282.24954432 928 96 741.7504576 96 512 96 282.24954432 282.24954432 96 512 96ZM544 736L480 736 480 800 544 800 544 736ZM512.2952448 256.64610496C441.57726976 255.489344 377.22819968 301.1206464 357.66394688 370.52669568 354.01781184 383.46170624 352.35546368 401.42816192 352.35546368 415.98780736L352.35546368 415.98780736 416.35546368 415.98780736 416.56729536 408.5808992C416.9809312 400.82021056 417.96989888 392.47948096 419.26346432 387.8904352 430.77765376 347.04275136 468.98897024 319.94628928 511.248512 320.63754496 564.72860672 321.51233984 607.554592 358.4738848 608.34222656 416.42270848 608.88517824 456.36943488 596.93322496 476.61043904 556.124528 517.76842944L556.124528 517.76842944 543.2981664 530.90088384C497.88380224 578.59564288 480 612.83118528 480 672.6246144L480 672.6246144 544 672.6246144 544.23439808 663.0609024C546.13973632 626.16948096 560.01985152 604.73470528 598.90074624 565.52317824L598.90074624 565.52317824 611.51250368 552.613136C655.9401472 505.9730656 673.1166976 472.96825088 672.3363136 415.55290688 671.047328 320.71799424 598.442304 258.05524608 512.2952448 256.64610496Z"
        fill-opacity=".9"
        fill-rule="evenodd"
      />
    </svg>
    
    用于帮助用户操作的信息提示
    
    <span
      class="t-message-close"
    >
      <!---->
    </span>
  </div>
  <div
    class="t-message t-is-loading"
  >
    <svg
      class="t-icon t-icon-loading"
      version="1.1"
      viewBox="0 0 1024 1024"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M96 512C96 282.264 282.264 96 512 96 741.736 96 928 282.264 928 512L824 512 824 512C824 339.68715776 684.3128448 200 512 200 339.68715776 200 200 339.68715776 200 512 200 684.3128448 339.68715776 824 512 824L512 928C282.264 928 96 741.736 96 512Z"
        fill-opacity=".9"
        fill-rule="evenodd"
      />
    </svg>
    
    用于表示操作正在生效的过程中
    
    <span
      class="t-message-close"
    >
      <!---->
    </span>
  </div>
</div>
`;
